zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1027,6 +1029,12 @@ private void validateCreateRequest(String path, CreateMode createMode, Request r
                         rv.add(new ACL(a.getPerms(), cid));
                     }
                 }
+                // If the znode path contains open read access node path prefix, add (world:anyone, r) in addition
+                if (ZNodeGroupAclProperties.getInstance().getOpenReadAccessPathPrefixes().stream()
+                    .anyMatch(path::contains)) {
```

Comment:
Question- is there a way we're using `contains()` here? Shouldn't it be `startsWith()` since it's a path prefix?